### PR TITLE
Matrix plot visualization

### PR DESCRIPTION
**Pull request by ianhelle**
Includes
- vis/matrix_plot.py - main plotting code
- MatrixPlot.ipynb - notebook
- visualization/MatrixPlot.rst - documentation
- tests/vis/test_matrix_plot.py - unit test
- mp_pandas_plot - generic pandas extension "mp_plot" that includes all
  visualizations

### CODE REVIEW

File: docs/source/visualization/MatrixPlot.rst [hunk 1]
```diff
@@ -0,0 +1,271 @@
+Matrix Plot
+===========
+
+The matrix plot is designed to show interactions between two sets of items
+(columns in a pandas DataFrame) in a x-y grid.
+
+For example, if you have a DataFrame with source and destination
+IP addresses (a firewall log), you can plot the source IPs
```
Comment (petebryan) [inline docs/source/visualization/MatrixPlot.rst:8]: (i.e. a firewall log)

File: msticpy/vis/matrix_plot.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    plot.xaxis.axis_label = param.x_column
+    plot.yaxis.axis_label = param.y_column
+
+    if not param.hide:
+        show(plot)
+    return plot
+
+
+def _set_plot_params(plot):
+    plot.title.text_font_size = "15pt"
```
Comment (petebryan) [inline msticpy/vis/matrix_plot.py:250]: Could consider making this variable based on font size parameter.

**Review by petebryan — APPROVED** [review]

File: docs/source/visualization/MatrixPlot.rst (hunk 1 above)
Comment (ianhelle) [reply docs/source/visualization/MatrixPlot.rst:8]: Done

File: msticpy/vis/matrix_plot.py (hunk 2 above)
Comment (ianhelle) [reply msticpy/vis/matrix_plot.py:250]: done - now have font_size as well as max_label_font_size

**Comment by ianhelle** [comment]
Already reviewed by Pete. Admin merge to override re-review.

**Pull request merged**